roles/test_operator/defaults/main.yml

```diff
@@ -40,6 +40,32 @@ cifmw_test_operator_tempest_image: "{{ cifmw_test_operator_tempest_registry }}/{
 cifmw_test_operator_tempest_image_tag: current-podified
 cifmw_test_operator_tempest_tests_include_override_scenario: false
 cifmw_test_operator_tempest_tests_exclude_override_scenario: false
+
+# Enabling SRBAC by default, in jobs where this does not make sense should be turned off explicitly
+cifmw_tempest_tempestconf_config_defaults:
+  deployerInput: |
+    [enforce_scope]
+    nova = true
```

Comment:
scope enforcemt is slightly different as almost all project removed scopes other then project scope as the desgin fo system scope fundementally broke many workflows.

so nova for example only uses project scope

scope are similar to using the new defaults in that really you should not mix an match enforcement in differnt services.

its technially possible to do but i dont think it is correct to do.

Reply:
hmm, i thought enforce scope means srbac, if not, than this change isn't needed as there is no other setting in tempest for srbac functionality

Reply:
Isn't it about setting the enforce_scope.<service>=True enabled? 
I see that SRBAC tests relies on that, so shouldn't we enable it?
For example https://opendev.org/openstack/glance-tempest-plugin/src/branch/master/glance_tempest_plugin/tests/rbac/v2/base.py#L32

Reply:
no unless tempest ahve really badly named there tests.

enforce_new_defaults = true enables the new rbac policies that provide the reader member, admin  and service user personas


enforce_scope = true enabled enforcement of system vs project scopted apis.

as afar as I'm aware scopes are only used by keystone and ironic.

nova and placement deleted all scopes other then project scope

we should have both 
enforce_new_defaults and enforce_scope set in all operators at this point
but enforce_new_defaults is the primary part of srbac.

if you not familiar with why scopes were removed from most of the projects following
the zed ptg

https://governance.openstack.org/tc/goals/selected/consistent-and-secure-rbac.html#direction-change

Reply:
jsut to be clear all apis have scopes_enforcement and the new defaults enabled
only ironic an keyston use a scope other the project scope in there policy.
and we do not plan to add any additional usage fo scopes like system scope or domain scope.

the primary features of SRBAC are related to the enforce_new_defaults which enable the use of the admin, member, reader and service roles in the policies.

Reply:
I've followed the story, but I think it may be, as you suggested, a case of bad naming for some tests, and I'd advise people to check that before just dismissing that option as not needed (for sure we wouldn't run relevant glance tests).

Reply:
I think we do need `[oslo_policy] enforce_scope = True` in every service for Secure RBAC to be fully enforced.  An important part of the direction change in the RBAC spec is this guidance:

> Services with project resources that have already implemented scope (or have yet to) should make all policy rules set `scope_types=[‘project’]`

Keystone will be issuing tokens with `project`, `domain`, and `system` scopes, and the `enforce_scope` option in oslo.policy ensures that tokens with scopes that are not part of the `scope_type` list for a specific policy will get rejected before the policy is enforced.  For projects that have not added `scope_types=[...]` to their policies, the enforce_scope option is basically a no-op, but we probably still need to test to make sure that requests with scopes other than `'project'` do get rejected by those APIs with a 403.

Some policies might reject a request with the wrong scope e.g. if the policy checks the token's project_id against a target.project_id. However, other policies might not be able to reject a request and return errors instead.  e.g. a request to `POST /v1/some_entity` that assigns ownership based on the context project_id might error out with a domain-scoped token.

Reply:
ack, thanks, so do we agree that defaults look ok as they are now so that we can proceed with this?

Reply:
the assertion i was orignaly making is not that we should not be enabling scope enforcement it was that 99% for the srbac feature is implemented by enforce_new_defaults

so if you only enabled enforce_scope then the reader role would not work for example.
it would have fully write access

almost none of the project work with domain or system scoped tokens even without 
enforce_scope (only keystone and ironic support scopes other then project) but you should enable both options in the service it you want SRBAC to be enabled.

in the tempestconf I'm not sure if there is 1 or 2 options but
we have been defaultign to srbac in the operator for some time now so we are cornfiug the service to use the srbac functionally today.

Reply:
Some plugins did the wrong thing, so that option, even if the name is wrong, enables the correct SRBAC tests. I think the confusion comes from that. The name is wrong but we need to set these options.

Reply:
> Some plugins did the wrong thing, so that option, even if the name is wrong, enables the correct SRBAC tests. I think the confusion comes from that. The name is wrong but we need to set these options.

so this is the sailiant point 

https://github.com/openstack/tempest/blob/master/tempest/config.py#L1213-L1219
the config option states that option also enables testing of enforce_new_defaults

which is the primary feature in the SRBAC community goal

so this should be set to true for nova too.

and an upstream tempest bug should be filed to fix the name of the config option
given the current one is very misleading.